File: backend/WholesaleERP.API/Models/InvoiceItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WholesaleERP.API.Models
{
    public class InvoiceItem
    {
        [Key]
        public int Id { get; set; }

        // Foreign Keys
        [Required]
        public int InvoiceId { get; set; }

        [Required]
        public int ItemId { get; set; }

        [Required]
        [StringLength(255)]
        public string ItemName { get; set; } = string.Empty; // Store name at time of invoice

        [Column(TypeName = "decimal(18,2)")]
        public decimal Quantity { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal UnitPrice { get; set; }

        [Column(TypeName = "decimal(5,2)")]
        public decimal DiscountPercentage { get; set; } = 0;

        [Column(TypeName = "decimal(18,2)")]
        public decimal DiscountAmount { get; set; } = 0;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Total { get; set; }

        [StringLength(50)]
        public string? Unit { get; set; }

        // Navigation Properties
        public virtual Invoice Invoice { get; set; } = null!;
        public virtual Item Item { get; set; } = null!;

        // Computed property
        [NotMapped]
        public decimal LineTotal => (Quantity * UnitPrice) - DiscountAmount;
    }
}

File: backend/src/entities/InvoiceItem.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn } from 'typeorm';
import { Invoice } from './Invoice';
import { Item } from './Item';

@Entity('invoice_items')
export class InvoiceItem {
  @PrimaryGeneratedColumn()
  id: number;

  @ManyToOne(() => Invoice, invoice => invoice.items, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'invoiceId' })
  invoice: Invoice;

  @Column()
  invoiceId: number;

  @ManyToOne(() => Item, item => item.invoiceItems)
  @JoinColumn({ name: 'itemId' })
  item: Item;

  @Column()
  itemId: number;

  @Column({ type: 'varchar', length: 255 })
  itemName: string; // Store name at time of invoice

  @Column({ type: 'decimal', precision: 10, scale: 2 })
  quantity: number;

  @Column({ type: 'decimal', precision: 10, scale: 2 })
  unitPrice: number;

  @Column({ type: 'decimal', precision: 5, scale: 2, default: 0 })
  discountPercentage: number;

  @Column({ type: 'decimal', precision: 10, scale: 2, default: 0 })
  discountAmount: number;

  @Column({ type: 'decimal', precision: 10, scale: 2 })
  total: number;

  @Column({ type: 'varchar', length: 50, nullable: true })
  unit: string;

  // Virtual property to calculate line total
  get lineTotal(): number {
    const subtotal = this.quantity * this.unitPrice;
    return subtotal - this.discountAmount;
  }
}